tests/golden/copy-command/copy-command.bats

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bats
+
+load '../helpers/bats-support/load'
+load '../helpers/bats-assert/load'
+load '../helpers'
+
+@test "copy an entry" {
+  run coffer create /a
+  assert_output "[SUCCESS] Entry created at '/a'."
+
+  run coffer copy /a /b
+  assert_output "[SUCCESS] Copied '/a' to '/b'."
+
+  run cleanOutput coffer view /
+  cat copy-an-entry.gold | assert_output -
+}
+
+@test "copy a directory" {
+  run coffer create /a/b/c
+  assert_output "[SUCCESS] Entry created at '/a/b/c'."
+
+  run coffer copy /a /
+  assert_output "[SUCCESS] Copied '/a/b/c' to '/b/c'."
+
+  run cleanOutput coffer view /
+  cat copy-a-directory.gold | assert_output -
+}
+
+@test "copy one directory to other" {
+  run coffer create /a/b/c
+  assert_output "[SUCCESS] Entry created at '/a/b/c'."
+
+  run coffer create /c/d/e
+  assert_output "[SUCCESS] Entry created at '/c/d/e'."
+
+  run coffer copy /a /c
+  assert_output "[SUCCESS] Copied '/a/b/c' to '/c/b/c'."
+
+  run cleanOutput coffer view /
+  cat copy-one-directory-to-other.gold | assert_output -
+}
+
+@test "trying to copy an entry to directory" {
+  run coffer create /a
+  assert_output "[SUCCESS] Entry created at '/a'."
+
+  run coffer create /b/c/d
+  assert_output "[SUCCESS] Entry created at '/b/c/d'."
+
+  run coffer copy /a /b/c
+
+  assert_failure
+  cat copy-an-entry-to-directory.gold | assert_output --partial -
+}
+
+@test "copy a directory without force" {
+  run coffer create /a/b/c
+  assert_output "[SUCCESS] Entry created at '/a/b/c'."
+
+  run coffer create /d/b/c
+  assert_output "[SUCCESS] Entry created at '/d/b/c'."
+
+  run coffer copy /a /d
+
+  assert_failure
+  cat copy-a-directory-without-force.gold | assert_output --partial -
```

Comment:
This test tries to copy a directory with one entry, and that entry clashes with an existing entry.

To make this test more robust, we should try to copy a directory with 3 entries, where 2 of them clash and the 3rd doesn't.

This way, we'll also be asserting that:
* coffer reports _all_ clashing entries (and not just the first clash).
* coffer _only_ reports clashing entries. Entries that can be safely copied do not show up in the output.

The same reasoning also applies to similar tests.

Reply:
Done